File: py/features.py
# GithubProjectIssues SDK feature factory

from feature.base_feature import GithubProjectIssuesBaseFeature
from feature.test_feature import GithubProjectIssuesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GithubProjectIssuesBaseFeature(),
        "test": lambda: GithubProjectIssuesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()

File: rb/features.rb
# GithubProjectIssues SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module GithubProjectIssuesFeatures
  def self.make_feature(name)
    case name
    when "base"
      GithubProjectIssuesBaseFeature.new
    when "test"
      GithubProjectIssuesTestFeature.new
    else
      GithubProjectIssuesBaseFeature.new
    end
  end
end
